prov/shm/src/smr.h

```diff
@@ -320,6 +319,20 @@ static inline bool smr_vma_enabled(struct smr_ep *ep,
 			peer_smr->xpmem_cap_self == SMR_VMA_CAP_ON);
 }
 
+static inline void smr_set_ipc_valid(struct smr_region *region, uint64_t id)
+{
+	if (ofi_hmem_is_initialized(FI_HMEM_ZE) &&
+	    region->map->peers[id].pid_fd == -1)
+		smr_peer_data(region)[id].ipc_invalid = 1;
```

Comment:
I'm going to be really annoying and recommend that we switch the bool from ipc_invalid to ipc_valid. Originally it made sense to do it as invalid so that it would get initialized to 0 and we wouldn't have to modify the checks to handle other ifaces but now that we check for ZE when we initialize the value, I think it makes more sense to have ipc_valid and then have this check be
```
static inline void smr_set_ipc_valid(struct smr_region *region, uint64_t id)
{
	if (ofi_hmem_is_initialized(FI_HMEM_ZE) &&
	    region->map->peers[id].pid_fd == -1)
		smr_peer_data(region)[id].ipc_valid = 0;
        else
               smr_peer_data(region)[id].ipc_valid = 1;
}
```
and then the smr_ipc_valid check can be simplified to remove the !s
That change makes the bool more readable and removes a few instructions on the fast path to check whether we can do ipc.
Sorryyyy